matchtree.go

```diff
@@ -983,14 +991,33 @@ func (d *indexData) newMatchTree(q query.Q, opt matchTreeOpt) (matchTree, error)
 		optCopy := opt
 		optCopy.DisableWordMatchOptimization = true
 
-		subMT, err := d.newMatchTree(s.Expr, optCopy)
+		anchored := false
+		expr := s.Expr
+		switch e := s.Expr.(type) {
+		case *query.Regexp:
+			pattern := e.Regexp.String()
+			if strings.HasPrefix(pattern, "^") && strings.HasSuffix(pattern, "$") {
```

Comment:
you can test this without creating a string first (look at the ast). Additionally because you do it this way I think things like case insenstive regex won't match since the string won't start with ^ then. Another benefit of doing it with the AST is it avoids extra work per shard.

Reply:
ah, good call. I took the easy path, but didn't think about things like `(?i)`. Fixed in the latest commit